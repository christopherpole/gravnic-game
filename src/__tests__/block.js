import { MOVE_UP, ENTITIES, changeGravityDirection } from '../index';

describe('Block entity', () => {
  it('Should match with other blocks', () => {
    const gameState = [
      [
        {
          staticEntity: { id: 2, entityId: ENTITIES.FLOOR },
          movableEntity: { id: 1, entityId: ENTITIES.BLOCK },
        },
      ],
      [{ staticEntity: { id: 3, entityId: ENTITIES.FLOOR }, movableEntity: null }],
      [
        {
          staticEntity: { id: 5, entityId: ENTITIES.FLOOR },
          movableEntity: { id: 4, entityId: ENTITIES.BLOCK },
        },
      ],
    ];

    const gameStates = changeGravityDirection(gameState, MOVE_UP);

    expect(gameStates).toMatchSnapshot();
  });

  it('Should wait until all blocks have stopped moving before matching', () => {
    const gameState = [
      [
        {
          staticEntity: { id: 2, entityId: ENTITIES.FLOOR },
          movableEntity: { id: 1, entityId: ENTITIES.BLOCK },
        },
      ],
      [{ staticEntity: { id: 3, entityId: ENTITIES.FLOOR }, movableEntity: null }],
      [
        {
          staticEntity: { id: 5, entityId: ENTITIES.FLOOR },
          movableEntity: { id: 4, entityId: ENTITIES.BLOCK },
        },
      ],
      [{ staticEntity: { id: 6, entityId: ENTITIES.FLOOR }, movableEntity: null }],
      [{ staticEntity: { id: 7, entityId: ENTITIES.FLOOR }, movableEntity: null }],
      [{ staticEntity: { id: 8, entityId: ENTITIES.FLOOR }, movableEntity: null }],
      [{ staticEntity: { id: 9, entityId: ENTITIES.FLOOR }, movableEntity: null }],
      [
        {
          staticEntity: { id: 11, entityId: ENTITIES.FLOOR },
          movableEntity: { id: 10, entityId: ENTITIES.BLOCK },
        },
      ],
    ];

    const gameStates = changeGravityDirection(gameState, MOVE_UP);

    expect(gameStates).toMatchSnapshot();
  });

  it('Should match more than two blocks', () => {
    const gameState = [
      [
        {
          staticEntity: { id: 2, entityId: ENTITIES.FLOOR },
          movableEntity: { id: 1, entityId: ENTITIES.BLOCK },
        },
        { staticEntity: null, movableEntity: null },
        {
          staticEntity: { id: 4, entityId: ENTITIES.FLOOR },
          movableEntity: { id: 3, entityId: ENTITIES.BLOCK },
        },
      ],
      [
        { staticEntity: { id: 5, entityId: ENTITIES.FLOOR }, movableEntity: null },
        {
          staticEntity: { id: 7, entityId: ENTITIES.FLOOR },
          movableEntity: { id: 6, entityId: ENTITIES.BLOCK },
        },
        { staticEntity: { id: 8, entityId: ENTITIES.FLOOR }, movableEntity: null },
      ],
      [
        { staticEntity: { id: 9, entityId: ENTITIES.FLOOR }, movableEntity: null },
        { staticEntity: { id: 10, entityId: ENTITIES.FLOOR }, movableEntity: null },
        { staticEntity: { id: 11, entityId: ENTITIES.FLOOR }, movableEntity: null },
      ],
      [
        { staticEntity: null, movableEntity: null },
        {
          staticEntity: { id: 13, entityId: ENTITIES.FLOOR },
          movableEntity: { id: 12, entityId: ENTITIES.BLOCK },
        },
        { staticEntity: null, movableEntity: null },
      ],
    ];

    const gameStates = changeGravityDirection(gameState, MOVE_UP);

    expect(gameStates).toMatchSnapshot();
  });
});
