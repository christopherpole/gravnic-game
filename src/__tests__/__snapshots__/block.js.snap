// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Block entity Should match more than two blocks 1`] = `
Array [
  Array [
    Array [
      Object {
        "movableEntity": Object {
          "entityId": "BLOCK",
          "id": 1,
        },
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 2,
        },
      },
      Object {
        "movableEntity": null,
        "staticEntity": null,
      },
      Object {
        "movableEntity": Object {
          "entityId": "BLOCK",
          "id": 3,
        },
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 4,
        },
      },
    ],
    Array [
      Object {
        "movableEntity": null,
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 5,
        },
      },
      Object {
        "movableEntity": Object {
          "entityId": "BLOCK",
          "id": 6,
        },
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 7,
        },
      },
      Object {
        "movableEntity": null,
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 8,
        },
      },
    ],
    Array [
      Object {
        "movableEntity": null,
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 9,
        },
      },
      Object {
        "movableEntity": Object {
          "entityId": "BLOCK",
          "id": 12,
        },
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 10,
        },
      },
      Object {
        "movableEntity": null,
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 11,
        },
      },
    ],
    Array [
      Object {
        "movableEntity": null,
        "staticEntity": null,
      },
      Object {
        "movableEntity": null,
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 13,
        },
      },
      Object {
        "movableEntity": null,
        "staticEntity": null,
      },
    ],
  ],
  Array [
    Array [
      Object {
        "movableEntity": Object {
          "entityId": "BLOCK",
          "id": 1,
        },
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 2,
        },
      },
      Object {
        "movableEntity": null,
        "staticEntity": null,
      },
      Object {
        "movableEntity": Object {
          "entityId": "BLOCK",
          "id": 3,
        },
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 4,
        },
      },
    ],
    Array [
      Object {
        "movableEntity": null,
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 5,
        },
      },
      Object {
        "movableEntity": Object {
          "entityId": "BLOCK",
          "fading": true,
          "id": 6,
        },
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 7,
        },
      },
      Object {
        "movableEntity": null,
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 8,
        },
      },
    ],
    Array [
      Object {
        "movableEntity": null,
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 9,
        },
      },
      Object {
        "movableEntity": Object {
          "entityId": "BLOCK",
          "fading": true,
          "id": 12,
        },
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 10,
        },
      },
      Object {
        "movableEntity": null,
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 11,
        },
      },
    ],
    Array [
      Object {
        "movableEntity": null,
        "staticEntity": null,
      },
      Object {
        "movableEntity": null,
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 13,
        },
      },
      Object {
        "movableEntity": null,
        "staticEntity": null,
      },
    ],
  ],
  Array [
    Array [
      Object {
        "movableEntity": Object {
          "entityId": "BLOCK",
          "id": 1,
        },
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 2,
        },
      },
      Object {
        "movableEntity": null,
        "staticEntity": null,
      },
      Object {
        "movableEntity": Object {
          "entityId": "BLOCK",
          "id": 3,
        },
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 4,
        },
      },
    ],
    Array [
      Object {
        "movableEntity": null,
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 5,
        },
      },
      Object {
        "movableEntity": null,
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 7,
        },
      },
      Object {
        "movableEntity": null,
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 8,
        },
      },
    ],
    Array [
      Object {
        "movableEntity": null,
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 9,
        },
      },
      Object {
        "movableEntity": null,
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 10,
        },
      },
      Object {
        "movableEntity": null,
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 11,
        },
      },
    ],
    Array [
      Object {
        "movableEntity": null,
        "staticEntity": null,
      },
      Object {
        "movableEntity": null,
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 13,
        },
      },
      Object {
        "movableEntity": null,
        "staticEntity": null,
      },
    ],
  ],
]
`;

exports[`Block entity Should match with other blocks of the same color 1`] = `
Array [
  Array [
    Array [
      Object {
        "movableEntity": Object {
          "color": "#ff0000",
          "entityId": "BLOCK",
          "id": 1,
        },
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 2,
        },
      },
    ],
    Array [
      Object {
        "movableEntity": Object {
          "color": "#ff0000",
          "entityId": "BLOCK",
          "id": 4,
        },
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 3,
        },
      },
    ],
    Array [
      Object {
        "movableEntity": null,
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 5,
        },
      },
    ],
  ],
  Array [
    Array [
      Object {
        "movableEntity": Object {
          "color": "#ff0000",
          "entityId": "BLOCK",
          "fading": true,
          "id": 1,
        },
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 2,
        },
      },
    ],
    Array [
      Object {
        "movableEntity": Object {
          "color": "#ff0000",
          "entityId": "BLOCK",
          "fading": true,
          "id": 4,
        },
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 3,
        },
      },
    ],
    Array [
      Object {
        "movableEntity": null,
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 5,
        },
      },
    ],
  ],
  Array [
    Array [
      Object {
        "movableEntity": null,
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 2,
        },
      },
    ],
    Array [
      Object {
        "movableEntity": null,
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 3,
        },
      },
    ],
    Array [
      Object {
        "movableEntity": null,
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 5,
        },
      },
    ],
  ],
]
`;

exports[`Block entity Should not match with other blocks that are not the same color 1`] = `
Array [
  Array [
    Array [
      Object {
        "movableEntity": Object {
          "color": "#00ff00",
          "entityId": "BLOCK",
          "id": 1,
        },
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 2,
        },
      },
    ],
    Array [
      Object {
        "movableEntity": Object {
          "color": "#ff0000",
          "entityId": "BLOCK",
          "id": 4,
        },
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 3,
        },
      },
    ],
    Array [
      Object {
        "movableEntity": null,
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 5,
        },
      },
    ],
  ],
]
`;

exports[`Block entity Should wait until all blocks have stopped moving before matching 1`] = `
Array [
  Array [
    Array [
      Object {
        "movableEntity": Object {
          "entityId": "BLOCK",
          "id": 1,
        },
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 2,
        },
      },
    ],
    Array [
      Object {
        "movableEntity": Object {
          "entityId": "BLOCK",
          "id": 4,
        },
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 3,
        },
      },
    ],
    Array [
      Object {
        "movableEntity": null,
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 5,
        },
      },
    ],
    Array [
      Object {
        "movableEntity": null,
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 6,
        },
      },
    ],
    Array [
      Object {
        "movableEntity": null,
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 7,
        },
      },
    ],
    Array [
      Object {
        "movableEntity": null,
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 8,
        },
      },
    ],
    Array [
      Object {
        "movableEntity": Object {
          "entityId": "BLOCK",
          "id": 10,
        },
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 9,
        },
      },
    ],
    Array [
      Object {
        "movableEntity": null,
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 11,
        },
      },
    ],
  ],
  Array [
    Array [
      Object {
        "movableEntity": Object {
          "entityId": "BLOCK",
          "id": 1,
        },
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 2,
        },
      },
    ],
    Array [
      Object {
        "movableEntity": Object {
          "entityId": "BLOCK",
          "id": 4,
        },
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 3,
        },
      },
    ],
    Array [
      Object {
        "movableEntity": null,
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 5,
        },
      },
    ],
    Array [
      Object {
        "movableEntity": null,
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 6,
        },
      },
    ],
    Array [
      Object {
        "movableEntity": null,
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 7,
        },
      },
    ],
    Array [
      Object {
        "movableEntity": Object {
          "entityId": "BLOCK",
          "id": 10,
        },
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 8,
        },
      },
    ],
    Array [
      Object {
        "movableEntity": null,
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 9,
        },
      },
    ],
    Array [
      Object {
        "movableEntity": null,
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 11,
        },
      },
    ],
  ],
  Array [
    Array [
      Object {
        "movableEntity": Object {
          "entityId": "BLOCK",
          "id": 1,
        },
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 2,
        },
      },
    ],
    Array [
      Object {
        "movableEntity": Object {
          "entityId": "BLOCK",
          "id": 4,
        },
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 3,
        },
      },
    ],
    Array [
      Object {
        "movableEntity": null,
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 5,
        },
      },
    ],
    Array [
      Object {
        "movableEntity": null,
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 6,
        },
      },
    ],
    Array [
      Object {
        "movableEntity": Object {
          "entityId": "BLOCK",
          "id": 10,
        },
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 7,
        },
      },
    ],
    Array [
      Object {
        "movableEntity": null,
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 8,
        },
      },
    ],
    Array [
      Object {
        "movableEntity": null,
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 9,
        },
      },
    ],
    Array [
      Object {
        "movableEntity": null,
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 11,
        },
      },
    ],
  ],
  Array [
    Array [
      Object {
        "movableEntity": Object {
          "entityId": "BLOCK",
          "id": 1,
        },
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 2,
        },
      },
    ],
    Array [
      Object {
        "movableEntity": Object {
          "entityId": "BLOCK",
          "id": 4,
        },
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 3,
        },
      },
    ],
    Array [
      Object {
        "movableEntity": null,
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 5,
        },
      },
    ],
    Array [
      Object {
        "movableEntity": Object {
          "entityId": "BLOCK",
          "id": 10,
        },
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 6,
        },
      },
    ],
    Array [
      Object {
        "movableEntity": null,
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 7,
        },
      },
    ],
    Array [
      Object {
        "movableEntity": null,
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 8,
        },
      },
    ],
    Array [
      Object {
        "movableEntity": null,
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 9,
        },
      },
    ],
    Array [
      Object {
        "movableEntity": null,
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 11,
        },
      },
    ],
  ],
  Array [
    Array [
      Object {
        "movableEntity": Object {
          "entityId": "BLOCK",
          "id": 1,
        },
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 2,
        },
      },
    ],
    Array [
      Object {
        "movableEntity": Object {
          "entityId": "BLOCK",
          "id": 4,
        },
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 3,
        },
      },
    ],
    Array [
      Object {
        "movableEntity": Object {
          "entityId": "BLOCK",
          "id": 10,
        },
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 5,
        },
      },
    ],
    Array [
      Object {
        "movableEntity": null,
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 6,
        },
      },
    ],
    Array [
      Object {
        "movableEntity": null,
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 7,
        },
      },
    ],
    Array [
      Object {
        "movableEntity": null,
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 8,
        },
      },
    ],
    Array [
      Object {
        "movableEntity": null,
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 9,
        },
      },
    ],
    Array [
      Object {
        "movableEntity": null,
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 11,
        },
      },
    ],
  ],
  Array [
    Array [
      Object {
        "movableEntity": Object {
          "entityId": "BLOCK",
          "fading": true,
          "id": 1,
        },
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 2,
        },
      },
    ],
    Array [
      Object {
        "movableEntity": Object {
          "entityId": "BLOCK",
          "fading": true,
          "id": 4,
        },
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 3,
        },
      },
    ],
    Array [
      Object {
        "movableEntity": Object {
          "entityId": "BLOCK",
          "fading": true,
          "id": 10,
        },
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 5,
        },
      },
    ],
    Array [
      Object {
        "movableEntity": null,
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 6,
        },
      },
    ],
    Array [
      Object {
        "movableEntity": null,
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 7,
        },
      },
    ],
    Array [
      Object {
        "movableEntity": null,
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 8,
        },
      },
    ],
    Array [
      Object {
        "movableEntity": null,
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 9,
        },
      },
    ],
    Array [
      Object {
        "movableEntity": null,
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 11,
        },
      },
    ],
  ],
  Array [
    Array [
      Object {
        "movableEntity": null,
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 2,
        },
      },
    ],
    Array [
      Object {
        "movableEntity": null,
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 3,
        },
      },
    ],
    Array [
      Object {
        "movableEntity": null,
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 5,
        },
      },
    ],
    Array [
      Object {
        "movableEntity": null,
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 6,
        },
      },
    ],
    Array [
      Object {
        "movableEntity": null,
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 7,
        },
      },
    ],
    Array [
      Object {
        "movableEntity": null,
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 8,
        },
      },
    ],
    Array [
      Object {
        "movableEntity": null,
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 9,
        },
      },
    ],
    Array [
      Object {
        "movableEntity": null,
        "staticEntity": Object {
          "entityId": "FLOOR",
          "id": 11,
        },
      },
    ],
  ],
]
`;
